File: src/main/java/net/nextpulse/jacumulus/requests/ListInvoicesRequest.java
package net.nextpulse.jacumulus.requests;

import javax.xml.bind.annotation.XmlElement;

/**
 * Request to list all invoices for a given contactId
 *
 * @see <a href="https://apidoc.sielsystems.nl/content/contact-list-incoming-invoices">the api docs for incoming</a>
 * @see <a href="https://apidoc.sielsystems.nl/content/contact-list-incoming-invoices">the api docs for outgoing</a>
 */
public class ListInvoicesRequest extends AcumulusRequest {
  
  @XmlElement(name = "contactid")
  private String contactId;
  
  public ListInvoicesRequest(String contactId) {
    this.contactId = contactId;
  }
  
  public String getContactId() {
    return contactId;
  }
  
  public void setContactId(String contactId) {
    this.contactId = contactId;
  }
}

File: src/main/java/net/nextpulse/jacumulus/requests/ListInvoicesRequest.kt
package net.nextpulse.jacumulus.requests

import javax.xml.bind.annotation.XmlElement
import javax.xml.bind.annotation.XmlRootElement

/**
 * Request to list all invoices for a given contactId
 *
 * @see [the api docs for incoming](https://apidoc.sielsystems.nl/content/contact-list-incoming-invoices)
 * @see [the api docs for outgoing](https://apidoc.sielsystems.nl/content/contact-list-outgoing-invoices)
 */
@XmlRootElement(name = "myxml")
data class ListInvoicesRequest(
        /**
         * Contact ID to retrieve all invoices for.
         */
        @get:XmlElement(name = "contactid")
        var contactId: String? = null
) : AcumulusRequest()
